src/lexer.coffee

```diff
@@ -196,6 +196,8 @@ exports.Lexer = class Lexer
          @tokens.length > 1 and @tokens[@tokens.length - 2][0] not in ['.', '?.', '@']
         @error "'#{prev[1]}' cannot be used as a keyword, or as a function call
         without parentheses", prev[2]
+      else if prev[0] is '.' and @tokens.length > 1 and @tokens[@tokens.length - 2][1] is 'new'
```

Comment:
Elsewhere in this section we check for the token tag, not its value. So in other words, `@tokens[@tokens.length - 2][0] is 'NEW'`

Reply:
@GeoffreyBooth looks like that doesn't uniquely identify it though, its `[0]` is not `NEW` but `UNARY`

So added a check for `UNARY` type but kept secondary check for `new` value

Reply:
Huh. Okay, I guess that works.